static/app/views/ddm/queryBuilder.tsx

```diff
@@ -88,116 +95,128 @@ export function QueryBuilder({
     }
   }, [isMetaLoading, displayedMetrics, metricsQuery.mri, onChange]);
 
+  const stringifiedMetricWidget = stringifyMetricWidget(metricsQuery);
+
   return (
     <QueryBuilderWrapper>
-      <QueryBuilderRow>
-        <WrapPageFilterBar>
-          <CompactSelect
-            searchable
-            sizeLimit={100}
-            triggerProps={{prefix: t('Metric'), size: 'sm'}}
-            options={displayedMetrics.map(metric => ({
-              label: mriMode ? metric.mri : formatMRI(metric.mri),
-              value: metric.mri,
-              trailingItems: mriMode ? undefined : (
-                <Fragment>
-                  <Tag tooltipText={t('Type')}>{getReadableMetricType(metric.type)}</Tag>
-                  <Tag tooltipText={t('Unit')}>{metric.unit}</Tag>
-                </Fragment>
-              ),
-            }))}
-            value={metricsQuery.mri}
-            onChange={option => {
-              const availableOps =
-                meta
-                  .find(metric => metric.mri === option.value)
-                  ?.operations.filter(isAllowedOp) ?? [];
+      {isEdit ? (
```

Comment:
nit: can we do an early return instead for better readability?

Reply:
sure thing https://github.com/getsentry/sentry/pull/62308/commits/85aa7db963fd939df2f0f216f9a8dd02c3e7c982 👍